wallet/common/symbol/src/api/HarvestingService.js

```diff
@@ -0,0 +1,198 @@
+				return {
+					status: HarvestingStatus.INACTIVE
+				};
+			}
+		}
+
+		// Fetch node info and its unlocked accounts.
+		try {
+			const nodeInfoEndpoint = `${statisticsServiceUrl}/nodes/nodePublicKey/${nodePublicKey}`;
+			const nodeInfo = await this.#makeRequest(nodeInfoEndpoint);
+			const nodeUrl = nodeInfo.apiStatus.restGatewayUrl;
+			const unlockedAccountsEndpoint = `${nodeUrl}/node/unlockedaccount`;
+			const { unlockedAccount } = await this.#makeRequest(unlockedAccountsEndpoint);
+			const isAccountUnlocked = unlockedAccount.find(publicKey => publicKey === linkedPublicKey);
+
+			// Active status if all keys linked and the account is in the node's unlocked list.
+			// Pending status if all keys linked but the account is not in the node's unlocked list.
+			return {
+				status: isAccountUnlocked ? HarvestingStatus.ACTIVE : HarvestingStatus.PENDING,
+				nodeUrl
+			};
+		} catch {
+			// Inactive status if node isn't found in the statistics service (treat it as down).
+			return {
+				status: HarvestingStatus.NODE_UNKNOWN
+			};
+		}
+	};
+
+	/**
+	 * Fetches the harvesting summary of an account.
+	 * @param {NetworkProperties} networkProperties - Network properties.
+	 * @param {string} address - Requested account address.
+	 * @returns {Promise<HarvestingSummary>} - The harvesting summary.
+	 */
+	fetchSummary = async (networkProperties, address) => {
+		const chainHeight = await this.#api.network.pingNode(networkProperties.nodeUrl);
+
+		const dayInSeconds = 86400;
+		const dayInBlocks = Math.round(dayInSeconds / networkProperties.blockGenerationTargetTime) * 30;
+		const heightDayAgo = chainHeight - dayInBlocks;
+
+		let isLastPage = false;
+		let isEndBlockFound = false;
+		let pageNumber = 1;
+		let harvestedBlocks = [];
+
+		while (!isLastPage && !isEndBlockFound) {
+			const harvestedBlocksPage = await this.fetchHarvestedBlocks(networkProperties, address, { pageNumber });
+			const filteredPerDayPage = harvestedBlocksPage.filter(block => block.height >= heightDayAgo);
+			isLastPage = harvestedBlocksPage.length === 0;
+			isEndBlockFound = filteredPerDayPage.length !== harvestedBlocksPage.length;
+			harvestedBlocks.push(...filteredPerDayPage);
+			pageNumber++;
+		}
+
+		let latestBlockDate = null;
+		if (harvestedBlocks.length) {
+			const { height } = harvestedBlocks[0];
+			const endpoint = `${networkProperties.nodeUrl}/blocks/${height}`;
+			const { block } = await this.#makeRequest(endpoint);
+			const timestamp = parseInt(block.timestamp);
+
+			latestBlockDate = networkTimestampToUnix(timestamp, networkProperties.epochAdjustment);
+		}
+
+		return {
+			latestAmount: harvestedBlocks.length ? harvestedBlocks[0].amount.toFixed(2) : 0,
+			latestHeight: harvestedBlocks.length ? harvestedBlocks[0].height : null,
+			latestDate: latestBlockDate,
+			amountPer30Days: harvestedBlocks.length ? _.sumBy(harvestedBlocks, 'amount').toFixed(2) : 0,
```

Comment:
I thought it was only checking harvesting in the past day? Or should ``heightDayAgo`` be ``height30DayAgo``? 🤔